pkg/auth/serviceaccount.go

```diff
@@ -94,3 +97,27 @@ func EnsureOIDCServiceAccountExistsForResource(ctx context.Context, serviceAccou
 
 	return nil
 }
+
+type OIDCIdentityStatusMarker interface {
+	MarkOIDCIdentityCreatedSucceeded()
+	MarkOIDCIdentityCreatedSucceededWithReason(reason, messageFormat string, messageA ...interface{})
+	MarkOIDCIdentityCreatedFailed(reason, messageFormat string, messageA ...interface{})
+}
+
+func SetupOIDCServiceAccount(flags feature.Flags, ctx context.Context, serviceAccountLister corev1listers.ServiceAccountLister, kubeclient kubernetes.Interface, gvk schema.GroupVersionKind, objectMeta metav1.ObjectMeta, marker OIDCIdentityStatusMarker, setAuthStatus func(a *duckv1.AuthStatus)) pkgreconciler.Event {
```

Comment:
@xiangpingjiang would you be able to add a unit test or two for this function? That way we can catch any breaking changes earlier if this every gets updated

Reply:
I will try to do that. SetupOIDCServiceAccount func rely on  EnsureOIDCServiceAccountExistsForResource func, the EnsureOIDCServiceAccountExistsForResource didn't have unit test. Do I need add unit test for EnsureOIDCServiceAccountExistsForResource first ?

Reply:
I think just `SetupOIDCServiceAccount` is totally fine, but if you wanted to add tests for both that would be great (just not necessary IMO)

Reply:
Hello @Cali0707 
I added some unit test, please give me a review if you have time